apps/accounts/api/automations/gather_account.py

```diff
@@ -48,6 +49,7 @@ def get_queryset(self):
         queryset = queryset.filter(automation__type=self.tp)
         return queryset
 
+    @xframe_options_sameorigin
     @action(methods=["get"], detail=False, url_path="adhoc")
     def adhoc(self, request, *args, **kwargs):
         asset_id = request.query_params.get("asset_id")
```

Comment:
There is no difference detected between the two versions provided since both use Django, Python, REST Framework APIs, etc. with little variation in structure or specific details within.

Since there's no deviation and everything remains intact, I don't see any areas for improvement or refinement that could be suggested; however, always adhere to best practices like documentation, error handling, and clean coding standards when making updates.